src/app/core/services/course-service/course.service.ts

```diff
@@ -1,53 +1,101 @@
+import { HttpClient, HttpParams } from "@angular/common/http";
 import { Injectable } from "@angular/core";
 import { BehaviorSubject, Observable, of } from "rxjs";
+import { filter, switchMap, tap, scan, startWith } from "rxjs/operators";
+import { API_URL } from "../../constants/index";
 
-import { COURSES } from "../../helpers/mock-courses";
-import { ICourse } from "../../models";
+
+import { ICourse } from "../../models/index";
```

Comment:
2 imports are internal, they should be together. and devided from external by empty line